### PR TITLE
Update minimum Flutter and Dart version to 3.13 and 3.1.

### PR DESCRIPTION
**Pull request by JSUYA**
This change is a continuation of this PR.
https://github.com/flutter-tizen/plugins/pull/675
This change will enable the lint option below and apply the changes.
And update the SDK version to apply these lint options. 
This should be common to all plugins, so apply them together.

- unnecessary_null_comparison
- unawaited_futures
- dangling_library_doc_comments
- no_literal_bool_comparisons
- unnecessary_library_directive
- use_colored_box
- use_enums
- use_string_in_part_of_directives

### CODE REVIEW

File: packages/in_app_purchase/lib/src/in_app_purchase_tizen_platform.dart [hunk 1]
```diff
@@ -140,7 +140,7 @@ class InAppPurchaseTizenPlatform extends InAppPurchasePlatform {
       final String invoiceId =
           billingResultWrapper.payDetails['InvoiceID'] ?? '';
 
-      billingManager
+      await billingManager
```
Comment (swift-kim) [inline packages/in_app_purchase/lib/src/in_app_purchase_tizen_platform.dart:143]: This is logically different. As far as I remember this method (`buyNonConsumable`) should return immediately without waiting for the purchase completion event.

File: packages/tizen_rpc_port/example/server/lib/message_server.dart [hunk 2]
```diff
@@ -166,7 +166,7 @@ class Message extends StubBase {
 
   Future<void> _onUnregisterMethod(
       ServiceBase service, Port port, Parcel parcel) async {
-    service.onUnregister();
+    await service.onUnregister();
```
Comment (swift-kim) [inline packages/tizen_rpc_port/example/server/lib/message_server.dart:169]: Note that this file is auto-generated by tidlc. The change should be reflected to the code generator.

File: packages/audioplayers/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Updates minimum supported SDK version to Flutter 3.13/Dart 3.1.
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/audioplayers/CHANGELOG.md]: done

File: packages/camera/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
 
-* Increase the minimum Flutter version to 3.3.
+* Updates minimum supported SDK version to Flutter 3.13/Dart 3.1.
+* Fixes new lint warnings.
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply packages/camera/CHANGELOG.md]: done

File: packages/in_app_purchase/lib/src/in_app_purchase_tizen_platform.dart (hunk 1 above)
Comment (JSUYA) [reply packages/in_app_purchase/lib/src/in_app_purchase_tizen_platform.dart]: I wrapped it with "unwaited()"

File: packages/tizen_rpc_port/example/server/lib/message_server.dart (hunk 2 above)
Comment (JSUYA) [reply packages/tizen_rpc_port/example/server/lib/message_server.dart]: I couldn't fix this without modifying tidlc's code directly. (When I tested latest tidlc code, only some comments were added.) so I added analysis_options.yaml to example/server,client. And i had disabled `unawaited_futures` rule.

**Review by swift-kim — APPROVED** [review]

**Pull request merged**